pkg/acme/webhook/apis/acme/v1alpha1/types.go

```diff
@@ -0,0 +1,114 @@
+	// The UID is meant to track the round trip (request/response) between the KAS and the WebHook, not the user request.
+	// It is suitable for correlating log entries between the webhook and apiserver, for either auditing or debugging.
+	UID types.UID `json:"uid"`
+
+	// Action is one of 'present' or 'cleanup'.
+	// If the action is 'present', the record will be presented with the
+	// solving service.
+	// If the action is 'cleanup', the record will be cleaned up with the
+	// solving service.
+	Action ChallengeAction `json:"action"`
+
+	// Type is the type of ACME challenge.
+	// Only dns-01 is currently supported.
+	Type string `json:"type"`
+
+	// DNSName is the name of the domain that is actually being validated, as
+	// requested by the user on the Certificate resource.
+	DNSName string `json:"dnsName"`
+
+	// Key is the key that should be presented.
+	// This key will already be signed by the account that owns the challenge.
+	// For DNS01, this is the key that should be set for the TXT record for
+	// ResolveFQDN.
+	Key string `json:"key"`
+
+	// ResourceNamespace is the namespace containing resources that are
+	// referenced in the providers config.
+	// If this request is solving for an Issuer resource, this will be the
+	// namespace of the Issuer.
+	// If this request is solving for a ClusterIssuer resource, this will be
+	// the configured 'cluster resource namespace'
+	ResourceNamespace string `json:"resourceNamespace"`
+
+	// ResolvedFQDN is the fully-qualified domain name that should be
+	// updated/presented after resolving all CNAMEs.
+	// This should be honoured when using the DNS01 solver type **only**
+	// +optional
+	ResolvedFQDN string `json:"resolvedFQDN,omitempty"`
+
+	// ResolvedZone is the zone encompassing the ResolvedFQDN.
+	// This is included as part of the ChallengeRequest so that webhook
+	// implementers do not need to implement their own SOA recursion logic.
+	// This indicates the zone that the provided FQDN is encompassed within,
+	// determined by performing SOA record queries for each part of the FQDN
+	// until an authoritative zone is found.
+	ResolvedZone string `json:"resolvedZone,omitempty"`
+
+	// AllowAmbientCredentials advises webhook implementations that they can
+	// use 'ambient credentials' for authenticating with their respective
+	// DNS provider services.
+	// This field SHOULD be honoured by all DNS webhook implementations, but
+	// in certain instances where it does not make sense to honour this option,
+	// an implementation may ignore it.
+	AllowAmbientCredentials bool `json:"allowAmbientCredentials"`
+
+	// Config contains unstructured JSON configuration data that the webhook
+	// implementation can unmarshal in order to fetch secrets or configure
+	// connection details etc.
+	// Secret values should not be passed in this field, in favour of
+	// references to Kubernetes Secret resources that the webhook can fetch.
+	// +optional
+	Config *apiext.JSON `json:"config,omitempty"`
```

Comment:
We may want to consider copying this type into our own repo at some point, to save people having to depend on apiextensions-apiserver.